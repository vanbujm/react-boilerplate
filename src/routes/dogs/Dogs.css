.root {
  display: flex;
  flex: 1;
  flex-direction: column;
}

.container {
  display: flex;
  flex-direction: column;
  margin: 0 auto;
}

.subCategory {
  margin-left: 1rem;
}
