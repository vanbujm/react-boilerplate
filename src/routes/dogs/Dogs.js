/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import { compose } from 'recompose';
import { graphql } from 'react-apollo';
import DogsQuery from './dogs.graphql';
import s from './Dogs.css';

export class DogsComponent extends React.Component {
  static propTypes = {
    data: PropTypes.shape({
      dogs: PropTypes.arrayOf(
        PropTypes.shape({
          name: PropTypes.string.isRequired,
          breed: PropTypes.string.isRequired,
          isGoodDog: PropTypes.string.isRequired,
        }),
      ),
    }),
  };

  render() {
    const dogMapper = dog => {
      const { id, name, breed, isGoodDog } = dog;
      return (
        <div key={id}>
          <h3>Name: {name}</h3>
          <p className={s.subCategory}>Breed: {breed}</p>
          <p className={s.subCategory}>Is a good dog?: {isGoodDog}</p>
        </div>
      );
    };
    const dogs = this.props.data.dogs
      ? this.props.data.dogs.map(dogMapper)
      : null;
    return (
      <div className={s.root}>
        <div className={s.container}>
          <h1>Dogs</h1>
          {dogs}
        </div>
      </div>
    );
  }
}

DogsComponent.defaultProps = {
  data: {
    dogs: [],
  },
};

export default compose(graphql(DogsQuery), withStyles(s))(DogsComponent);
