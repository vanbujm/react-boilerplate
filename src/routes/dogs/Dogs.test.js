/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import { componentWithMockContext } from '../../../test/testHelpers';
import { DogsComponent } from './Dogs';

const mockDogs = [
  {
    id: 1,
    name: 'Chewer',
    breed: 'Big dog',
    isGoodDog: 'Yes',
  },
  {
    id: 2,
    name: 'Nibbler',
    breed: 'Smol dog',
    isGoodDog: 'Yes',
  },
];

describe('Dogs Page', () => {
  test('renders', () => {
    const renderedComponent = renderer
      .create(componentWithMockContext()(<DogsComponent data={{ dogs: [] }} />))
      .toJSON();

    expect(renderedComponent).toMatchSnapshot();
  });

  test('renders dogs', () => {
    const renderedComponent = renderer
      .create(
        componentWithMockContext()(<DogsComponent data={{ dogs: mockDogs }} />),
      )
      .toJSON();

    expect(renderedComponent).toMatchSnapshot();
  });
});
