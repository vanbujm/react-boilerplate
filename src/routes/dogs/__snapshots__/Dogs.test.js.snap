// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dogs Page renders 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <h1>
      Dogs
    </h1>
  </div>
</div>
`;

exports[`Dogs Page renders dogs 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <h1>
      Dogs
    </h1>
    <div>
      <h3>
        Name: 
        Chewer
      </h3>
      <p
        className="subCategory"
      >
        Breed: 
        Big dog
      </p>
      <p
        className="subCategory"
      >
        Is a good dog?: 
        Yes
      </p>
    </div>
    <div>
      <h3>
        Name: 
        Nibbler
      </h3>
      <p
        className="subCategory"
      >
        Breed: 
        Smol dog
      </p>
      <p
        className="subCategory"
      >
        Is a good dog?: 
        Yes
      </p>
    </div>
  </div>
</div>
`;
